### DISCUSSION TITLE
Should we release 0.71.9?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#69] [Patches]
### released, please refer to #70 

---

Current Release: [0.71.8](https://github.com/facebook/react-native/releases/tag/v0.71.8)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

1. picked ~~https://github.com/facebook/react-native/commit/98009ad94b92320307f2721ee39dbeb9152c0a58~~
2. picked ~~https://github.com/facebook/react-native/commit/012e4bd654f1eee2b00a066ba50a7f9c44cc305b~~
3. https://github.com/facebook/react-native/pull/37452 - https://github.com/facebook/react-native/commit/c56968c5ef135bf18f41ab9fa65f0faa69af6729
5. picked ~~https://github.com/facebook/react-native/commit/2558c3d4f56776699602b116aff8c22b8bfa176a~~
6. picked ~~https://github.com/facebook/react-native/commit/94356e14ec0562a1fd5a208d93021f102ba9565e~~
7. Hermes side bump
    * with https://github.com/facebook/hermes/commit/f2f1a2e38e86632f24176d9e607c06507f900496 (ran into a conflict, need help from Hermes https://github.com/facebook/hermes/pull/1023)

#### Local commits to backport to main

1.

**Comment by kelset** [comment]
From previous patch discussion, from @leotm :
> could we pos bump [0.71-stable/sdks/.hermesversion](https://github.com/facebook/react-native/blob/0.71-stable/sdks/.hermesversion) hermes-2023-03-07-RNv0.71.4-31fdcf738940875c9bacf251e149006cf515d763
to address this Hermes crash https://github.com/facebook/hermes/issues/886 (so bump would include https://github.com/facebook/hermes/commit/f2f1a2e38e86632f24176d9e607c06507f900496 from 2023-03-23)

Further explanation:
> for our https://github.com/MetaMask/metamask-mobile/pull/6220 (wip) we [patched](https://github.com/MetaMask/metamask-mobile/pull/6220/commits/794e7c05e41e9ac52da67391b1ca5374579fc1d5) the lib, then later [updated](https://github.com/MetaMask/metamask-mobile/pull/6220/commits/5dc1949c5757f8a3e324ce004b24779cadb64ef0#diff-9a50ef936e92a26017fd739f42ee0a0e7f80aec39dafc0a13f27b2548ee5e51f) the proj to build from source with [custom](https://github.com/MetaMask/metamask-mobile/pull/6220/commits/8a5dce5433b0ca5fc5075e047b5d3e21c2ff2b56) Hermes build (0.71.6 tag + https://github.com/facebook/hermes/commit/f2f1a2e38e86632f24176d9e607c06507f900496 cherry-picked) and tested w/o the lib patch (gr8 success)
[removed](https://github.com/MetaMask/metamask-mobile/pull/6220/commits/8da441569ddbb283dca35f3e6b9b2997d9ee5c0f) it now and https://github.com/MetaMask/metamask-mobile/pull/6220#issuecomment-1540765257 to confirm

**Reply by kelset** [reply]
hey @leotm a quick update on this: we tried to cherry pick it but we hit a number of issues so we'll hold off from making this Hermes change.

**Reply by leotm** [reply]
np thx for the update ^ and appreciate the effort

**Comment by douglowder** [comment]
Once merged, please include https://github.com/facebook/react-native/pull/37445

**Comment by kelset** [comment]
https://github.com/facebook/react-native/pull/37452

**Comment by iceuponfire** [comment]
Please include https://github.com/facebook/react-native/pull/36067

**Comment by sunnylqm** [comment]
Enable RuntimeScheduler in old architecture 
https://github.com/facebook/react-native/commit/7211ef19624304b6a4d5219a8e0a2c67651b8b33
https://github.com/facebook/react-native/commit/133ccdcc67a7d19ffa5130949893c2792e3ad9fb

**Reply by Saadnajmi** [reply]
is this to fix performance regressions seen on the old architecture since 0.70/71 ish?

**Reply by cipolleschi** [reply]
Yea, we believe that this should be the fix. We would like to first double check that with RC.4. If confirmed, we will try to backport it to older versions, if feasible.

**Reply by Saadnajmi** [reply]
Thanks! We may also try locally testing this as well!

**Reply by fortmarek** [reply]
@cipolleschi has this regression been introduced in `0.71` or do we need to backport the fix to previous releases, too? It sounds like we'll need to backport it to 0.70 at least.

EDIT:
Based on [this](https://github.com/facebook/react-native/commit/133ccdcc67a7d19ffa5130949893c2792e3ad9fb) commit, the regression is there from 0.69.

**Reply by cortinico** [reply]
Hey all,
This is not easily backportable as we would also need to bump the `scheduler` version to the canary version we'll be using in 0.72.
We'll hold on backporting this for the time being. We first want to validate that the fix for 0.72 works for the majority of users and we'll consider eventually backporting it

**Comment by ste7en** [comment]
This adds `string` type to the `transform` style prop https://github.com/facebook/react-native/commit/2558c3d4f56776699602b116aff8c22b8bfa176a

**Comment by xsjcTony** [comment]
Please include https://github.com/facebook/react-native/commit/94356e14ec0562a1fd5a208d93021f102ba9565e
This fix the type for `autoComplete` prop of  `TextInput`

**Comment by lunaleaps** [comment]
I'm attempting to bump Hermes but ran into a conflict. Can someone from the Hermes team take a look?
https://github.com/facebook/hermes/pull/1023

**Reply by cipolleschi** [reply]
I took a stab at merging the conflict. The problem was that the PR updates a test file that was not there in 0.71-stable.
I saw that the PR is also introducing a new tests and updating a couple of older ones, so I decided to resolve the conflict by adding the modified test on the Repo. If the CI is happy, I think we are good to go!

**Reply by cipolleschi** [reply]
At the end, this change is incompatible with the status of Hermes of 0.71. It changes how some things behave and more tests and configurations are failing. We decided not to pick it. See [here](https://github.com/reactwg/react-native-releases/discussions/69#discussioncomment-6100211)

@lunaleaps do you think we can close the PR or do you still want for the Hermes team to have a look at it?

**Reply by leotm** [reply]
appreciate the effort ^ would be great if Hermes team could take a look

**Reply by lunaleaps** [reply]
From this [discussion](https://github.com/reactwg/react-native-releases/discussions/69#discussioncomment-6100211) we'll proceed 0.71.9 without a Hermes bump.
